torchmetrics/image_generation/fid.py

```diff
@@ -0,0 +1,322 @@
+                                            f" {allowed_input} but got {images_or_features}")
+        
+        self.images_or_features = images_or_features
+        self.perform_normalization = perform_normalization
+        self.normalizer = None
+        
+        if self.images_or_features == 'images' and not _TORCHVISION_AVAILABLE:
+            raise MisconfigurationException("FID metric requires torchvision to be installed for downloading"
+                                            " inception v3 network")
+        
+        if self.images_or_features == 'images':
+            if self.perform_normalization:
+                self.normalizer = transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
+            
+            self.inception = inception_v3(pretrained=True)
+            self.inception.fc = _Identity()
+            self.feature_size = 2048
+        else:
+            self.inception = None
+            self.feature_size = feature_size
+            
+        self.add_state("real_mean", torch.zeros(self.feature_size), dist_reduce_fx="mean")
+        self.add_state("real_cov", torch.zeros(self.feature_size, self.feature_size), dist_reduce_fx="mean")
+        self.add_state("real_nobs", torch.zeros(1), dist_reduce_fx="sum")
+        self.add_state("fake_mean", torch.zeros(self.feature_size), dist_reduce_fx="mean")
+        self.add_state("fake_cov", torch.zeros(self.feature_size, self.feature_size), dist_reduce_fx="mean")
+        self.add_state("fake_nobs", torch.zeros(1), dist_reduce_fx="sum")
+
+    def update(self, imgs_or_feature: torch.Tensor, real: bool = True) -> None:
+        """ Update distributions with new data
+        Args:
+            imgs: image data of shape ``[N, 3, 299, 299]``
+            real: boolian indicating if the images comes from the true data distribution or the fake data
+                distribution (generated images)
+        """
+        self._check_valid_input(imgs_or_feature)
+        
+        if self.images_or_features == 'images':
+            if self.perform_normalization:
+                imgs = self.normalizer(imgs_or_feature)
+            incep_score = self.inception(imgs)
+            
+        else:
+            incep_score = imgs_or_feature
+        
+        if real:
+            new_mean = _update_mean(self.real_mean, self.real_nobs, incep_score)
+            new_cov = _update_cov(self.real_cov, self.real_mean, new_mean, incep_score)
+            self.real_mean = new_mean
+            self.real_cov = new_cov
+            self.real_nobs += incep_score.shape[0]
+        else:
+            new_mean = _update_mean(self.fake_mean, self.fake_nobs, incep_score)
+            new_cov = _update_cov(self.fake_cov, self.fake_mean, new_mean, incep_score)
+            self.fake_mean = new_mean
+            self.fake_cov = new_cov
+            self.fake_nobs += incep_score.shape[0]
+
+    def compute(self) -> torch.Tensor:
```

Comment:
Note that after every .compute(), the real data statistics will be recomputed the next time you want to calculate the FID.
Is there a way to avoid that? This is an unnecessary overhead.

Thanks!